src/Common/src/Interop/Unix/System.Native/Interop.RealPath.cs

```diff
@@ -0,0 +1,44 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        /// <summary>
+        /// Takes a path containing relative subpaths or links and returns the absolute path.
+        /// </summary>
+        /// <param name="path">The input path containing relative subpaths or links</param>
+        /// <param name="resultPath">The output path; will be null on error</param>
+        /// <returns>Returns 0 on succes and -1 on failure; on failure, resultPath will be null</returns>
+        internal static int RealPath(string path, out string resultPath)
+        {
+            byte[] buffer = new byte[Interop.Sys.MaxPath];  // MaxPath should be big enough every time
```

Comment:
Ok.  In that case, SystemNative_RealPath can basically just do ```return realpath(path, nullptr);```, the DllImport just basically becomes ```string RealPath(string path, IntPtr nullResultPath)```, and the exposed wrapper basiclaly becomes ```return RealPath(path, IntPtr.Zero).```